core/chains/solana/soltxm/cache.go

```diff
@@ -0,0 +1,120 @@
+package soltxm
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/gagliardetto/solana-go"
+	solanaClient "github.com/smartcontractkit/chainlink-solana/pkg/solana/client"
+	"golang.org/x/exp/maps"
+)
+
+type TxCache struct {
```

Comment:
A cache is something that holds data, more often than not simple serializable data, which it tries to store in an efficient manner. It is usual for a cache to drop/invalidate some data (LRU for example) if it will make it more efficient.

Here we are storing references to objects that we absolutely need to keep, or we get memory leaks. And additionally, this component not only stores data but also manages a bunch of other components (kills goroutines & writes to Prom) and introduces side-effects (mutates them).

I'd suggest this is not really a "cache" but an aux (to a TXM) process manager object that manages goroutines (open subprocesses) + Prometheus (doing too much IMHO).